test/dsl/dsl_options.jl

```diff
@@ -152,6 +153,56 @@ let
     @test issetequal(parameters(rn5), @parameters k B X)
 end
 
+# Test whether user-defined functions are properly expanded in equations. 
+# TODO
+let
+    f(A, t) = 2*A*t
+
+    # Test user-defined function
+    rn = @reaction_network begin
+        @equations D(A) ~ f(A, t)
+    end
+    @test length(equations(rn)) == 1
+    @test equations(rn)[1] isa Equation
+    @species A(t)
+    @test isequal(equations(rn)[1], D(A) ~ 2*A*t)
+
+
+    # Test Catalyst function
+    rn2 = @reaction_network begin
+        @equations D(A) ~ hill(A, v, K, n)
+    end
+    @test length(equations(rn2)) == 1
+    @test equations(rn2)[1] isa Equation
+    @parameters K v n
+    @test isequal(Catalyst.expand_registered_functions(equations(rn2)[1]), D(A) ~ v*(A^n) / (A^n + K^n))
+
+
+    rn3 = @reaction_network begin
+        @species Iapp(t)
+        @equations begin
+            D(A) ~ Iapp
+            Iapp ~ f(A,t)
+        end
+    end
+    @test length(equations(rn3)) == 2
+    @test equations(rn3)[1] isa Equation
+    @test equations(rn3)[2] isa Equation
+    @variables Iapp(t)
+    @test isequal(equations(rn3)[1], D(A) ~ Iapp)
+    @test isequal(equations(rn3)[2], Iapp ~ 2*A*t)
+
+
+    g(A, K, n) = A^n + K^n
+    rn4 = @reaction_network begin
+        @equations D(A) ~ hill(A, v, K, n)*g(A, K, n)
+    end
+    @test length(equations(rn4)) == 1
+    @test equations(rn4)[1] isa Equation
+    @parameters v n
+    @test isequal(Catalyst.expand_registered_functions(equations(rn4)[1]), D(A) ~ v*(A^n))
+end
+
```

Comment:
Can we also add similar tests using a registered function for `f(A,t)`? (Just to make sure that case works too.)

Reply:
Ahh right, forgot about that! But maybe make one you explicitly create in case we change that in the future? (I've been debating whether we should stop registering those functions and just let MTK trace -- registration is currently breaking unit checks in them.)

Reply:
As in explicitly define the `hill` function inside the test block?

Reply:
Sure, but don't call it `hill`.